support/docker/build.sh

```diff
@@ -29,11 +29,136 @@ stderr.printf() {
+		PS3="Running as root. Do you want to continue?: "
+
+		select yn in "Yes" "No"; do
+			case $yn in
+				Yes) 
+					stderr.printf "\nProceeding as root user\n"
+					break
+					;;
+				No)
+					stderr.printf "\nQuiting the build script...\n"
+					exit 0
+					;;
+			esac
+		done
+	fi
+
+	# TODO: Add selection menu of available boards located in boards dir
+
+	# Find which container manager is available
+
+	declare -a availableManagers=()
+
+	# Check if docker is available
+	if ! command -v docker &> /dev/null;
+	then
+		stderr.printf "Docker not installed\n"
+	else
+		availableManagers+=('docker')
+	fi
+
+
+	# Check if podman is available
+	if ! command -v podman &> /dev/null;
+	then
+		stderr.printf "Podman not installed\n"
+	else
+		availableManagers+=('podman')
+	fi
+
+	# Exit if neither docker or podman is available
+	if [ ${#availableManagers[@]} -eq 0 ]
+	then
+		stderr.printf "\nCan't find docker or podman on your system.\nInstall one of them to continue...\n"
+		exit 1
+	fi
+
+	# Add exit command
+	availableManagers+=('Quit')
+	
+
+	local selectedManager
+	# Use switch statements to select from the available managers
+	stderr.printf "\nAvailable container managers on your system:\n"
+	PS3="Select one of the options or quit: "
+	select manager in ${availableManagers[*]}; do
+		case $manager in
+			docker|podman)
+				stderr.printf "\nSelected:\n $manager\n"
+				selectedManager=$manager
+				break
+				;;
+			Quit)
+				stderr.printf "\nSee you soon!\n"
+				exit 0	
+				;;
+			*)
+				stderr.printf "\nInvalid option\n"
+				;;
+		esac
+	done
+
+
+	# Check if the image is available
+	declare checkImg
+	case $selectedManager in
+		docker)
+			checkImg="$($selectedManager image ls -a nixos/nix | grep -Fq nixos/nix; echo $?)"
+			;;
+		podman)
+			checkImg=$($selectedManager image exists nix; echo $?)
+			;;
+		*)
+			stderr.printf "\nSomething went wrong when checking if the image is available with the chosen container manager: $selectedManager\n"
+			exit 1
+			;;
+	esac
+	
+	# Pull the image if it doesn't exists
+	if (( $checkImg == 0 ));
+	then
+		stderr.printf "\nNixOS image is present\n"
+	elif (( $checkImg == 1 ));
+	then
+		stderr.printf "\nCan\'t find NixOS image. Pulling...\n"
+		$selectedManager pull docker.io/nixos/nix
+	fi
+
+	# Check if the volume exists
 	local STORE_VOLUME
-	STORE_VOLUME="$(docker volume create "Tow-Boot--nix-store")"
+	declare volumeExists
+	case $selectedManager in
+		docker)
+			volumeExists=$($selectedManager volume ls | grep -Fq Tow-Boot--nix-store; echo $?)
+			;;
+		podman)
+			volumeExists=$($selectedManager volume exists Tow-Boot--nix-store; echo $?)
+			;;
+		*)
+			stderr.printf "\nSomething went wrong when checking if the volume exists with the chosen container manager: $selectedManager\n"
```

Comment:
```suggestion
			stderr.printf "\nUnexpected value for selectedManager ($selectedManager) when checking volume exists.\n"
```

This is a message that is unlikely to be seen by a user, but more likely by a developer adding a new manager. Let's make the message precise in what it is about.